ocaml-lsp-server/docs/ocamllsp/typeEnclosing-spec.md

```diff
@@ -0,0 +1,51 @@
+# Type Enclosing Request
+
+## Description
+
+Merlin has a concept of `type enclosing` that gets the type of ident under the
+cursor. It will highlight the ident and display its type. You can climb the
+typed-tree and display the type of bigger expressions surrounding the cursor.
+Since _LSP_ is stateless, the manipulation related to growing or shrinking
```

Comment:
Even if the interface we propose for `type-enclosing` is somehow stateless, LSP itself is not a stateless protocol.